File: backend/app/schemas/reports.py
from decimal import Decimal

from pydantic import BaseModel


class DashboardMetricCard(BaseModel):
    label: str
    value: int | Decimal | str
    secondary: str | None = None


class DashboardSummaryResponse(BaseModel):
    cards: list[DashboardMetricCard]
    latest_audit_events: int
    open_conversations: int
    pending_comments: int
    scheduled_posts: int
    current_token_balance: int
    monthly_token_credit: int


class TokenUsageSummaryResponse(BaseModel):
    total_tokens_consumed: int
    total_estimated_cost: Decimal
    total_action_count: int
    top_actions: list[dict]
    daily_usage: list[dict]


class BillingSummaryResponse(BaseModel):
    current_token_balance: int
    monthly_token_credit: int
    active_plan_code: str | None = None
    active_plan_name: str | None = None
    subscription_status: str | None = None
    billed_tokens: int
    estimated_llm_cost: Decimal
    estimated_action_cost: Decimal
    total_estimated_cost: Decimal
    llm_usage_by_model: list[dict]


class ConversationStatsResponse(BaseModel):
    total_conversations: int
    open_conversations: int
    resolved_conversations: int
    assigned_conversations: int
    paused_conversations: int
    recent_daily_counts: list[dict]


class CommentStatsResponse(BaseModel):
    total_comments: int
    pending_comments: int
    replied_comments: int
    flagged_comments: int
    need_review_comments: int
    recent_daily_counts: list[dict]


class PostStatsResponse(BaseModel):
    total_posts: int
    draft_posts: int
    scheduled_posts: int
    published_posts: int
    failed_posts: int
    recent_daily_counts: list[dict]

File: frontend/src/shared/types/reports.ts
export type DashboardMetricCard = {
  label: string;
  value: number | string;
  secondary: string | null;
};

export type DashboardSummary = {
  cards: DashboardMetricCard[];
  latest_audit_events: number;
  open_conversations: number;
  pending_comments: number;
  scheduled_posts: number;
  current_token_balance: number;
  monthly_token_credit: number;
};

export type TokenUsageSummary = {
  total_tokens_consumed: number;
  total_estimated_cost: string;
  total_action_count: number;
  top_actions: Array<{
    action_type: string;
    quantity: number;
    tokens_consumed: number;
    estimated_cost: string;
  }>;
  daily_usage: Array<{
    date: string;
    tokens_consumed: number;
  }>;
};

export type BillingSummary = {
  current_token_balance: number;
  monthly_token_credit: number;
  active_plan_code: string | null;
  active_plan_name: string | null;
  subscription_status: string | null;
  billed_tokens: number;
  estimated_llm_cost: string;
  estimated_action_cost: string;
  total_estimated_cost: string;
  llm_usage_by_model: Array<{
    model_name: string;
    total_tokens: number;
    estimated_cost: string;
  }>;
};

export type ConversationStats = {
  total_conversations: number;
  open_conversations: number;
  resolved_conversations: number;
  assigned_conversations: number;
  paused_conversations: number;
  recent_daily_counts: Array<{ date: string; count: number }>;
};

export type CommentStats = {
  total_comments: number;
  pending_comments: number;
  replied_comments: number;
  flagged_comments: number;
  need_review_comments: number;
  recent_daily_counts: Array<{ date: string; count: number }>;
};

export type PostStats = {
  total_posts: number;
  draft_posts: number;
  scheduled_posts: number;
  published_posts: number;
  failed_posts: number;
  recent_daily_counts: Array<{ date: string; count: number }>;
};
